src/scancode/api.py

```diff
@@ -288,29 +288,68 @@ def _licenses_data_from_match(
 
 
 def get_package_info(location, **kwargs):
+    """
+    Return a mapping of package information detected in the
+    file at `location`.
+
+    Note that all exceptions are caught if there are any errors while parsing a
+    package.
+    """
+    import warnings
+    warnings.warn(
+        "`get_package_info` is deprecated. Use `get_package_manifests` instead.",
+        DeprecationWarning,
+        stacklevel=1
+    )
+
+    from packagedcode.recognize import recognize_package_manifests
+    try:
+        recognized_package_manifests = recognize_package_manifests(location)
+        if recognized_package_manifests:
+            return dict(package_manifests=[
+                package_manifest.to_dict()
+                for package_manifest in recognized_package_manifests
+            ])
+    except Exception as e:
+        if TRACE:
+            logger.error('get_package_info: {}: Exception: {}'.format(location, e))
+
+        if SCANCODE_DEBUG_PACKAGE_API:
+            raise
+        else:
+            # attention: we are swallowing ALL exceptions here!
+            pass
+
+    return dict(package_manifests=[])
```

Comment:
Please keep the same API as before... the only addition to this function should be using the new recognize_package_manifests function and the deprecation warning.
It would be good to also add some  statement in the doc string that warns that the function is deprecated.